### PR TITLE
feat(chromecast): added tracks support, added session restoring support

### PR DESCRIPTION
**Pull request by jbreemhaar**

### CODE REVIEW

File: src/js/enableChromecast.js [hunk 1]
```diff
@@ -42,8 +43,8 @@ function onChromecastRequested(player) {
 }
 
 /**
- * Adds the Chromecast button to the player's control bar, if one does not already exist,
- * then starts listening for the `chromecastRequested` event.
+ * Adds the Chromecast bning foutton to the player's control bar, if one does not already
+ * exist, then starts lister the `chromecastRequested` event.
```
Comment (MathijsR94) [inline src/js/enableChromecast.js]: Unintentional change I guess?

File: src/js/enableChromecast.js [hunk 2]
```diff
@@ -42,8 +43,8 @@ function onChromecastRequested(player) {
 }
 
 /**
- * Adds the Chromecast button to the player's control bar, if one does not already exist,
- * then starts listening for the `chromecastRequested` event.
+ * Adds the Chromecast bning foutton to the player's control bar, if one does not already
```
Comment (sjors98) [inline src/js/enableChromecast.js]: Typo?

```suggestion
 * Adds the Chromecast button to the player's control bar, if one does not already
```

File: src/js/tech/ChromecastTech.js [hunk 3]
```diff
@@ -144,6 +152,45 @@ ChromecastTech = {
...
+   /**
+    * generate track
+    * @param trackData
+    */
+   generateTrack: function(trackData, id) {
+      var sub = new chrome.cast.media.Track(id, chrome.cast.media.TrackType.TEXT);
+
+      sub.trackContentId = trackData.src;
+      sub.trackContentType = 'text/vtt';
+      sub.subtype = chrome.cast.media.TextTrackType.SUBTITLES;
```
Comment (MathijsR94) [inline src/js/tech/ChromecastTech.js]: Is this always a subtitle?

File: src/js/tech/ChromecastTech.js [hunk 4]
```diff
@@ -144,6 +152,45 @@ ChromecastTech = {
...
+          media = castSession.getMediaSession(),
+          index, subtitles, tracksInfoRequest, i;
+
+
+      if (castSession) {
+         index = [];
+         subtitles = this.videojsPlayer.remoteTextTracks();
+         for (i = 0; i < subtitles.length; i++) {
+            if (subtitles[i].mode === 'showing') {
+               index = [ i ];
```
Comment (MathijsR94) [inline src/js/tech/ChromecastTech.js:199]: It's a style thing I know, but is feels like it should be this?
```suggestion
               index = [i];
```

File: src/js/tech/ChromecastTech.js [hunk 5]
```diff
@@ -181,20 +244,28 @@ ChromecastTech = {
...
+      }
+
+      this.trigger('loadstart');
+      this.trigger('loadeddata');
+      this.trigger('play');
+      this.trigger('playing');
+      this.videojsPlayer.hasStarted(true);
+      this._hasPlayedAnyItem = true;
+      this._isMediaLoading = false;
+      window.setTimeout(this._onPlayerStateChanged.bind(this), 1000); // magic number 1 second to validate playstate b/c race conditions
```
Comment (MathijsR94) [inline src/js/tech/ChromecastTech.js]: Not 100% sure, but should this timeout be cleaned up?

File: src/js/tech/ChromecastTech.js (hunk 4 above)
Comment (jbreemhaar) [reply src/js/tech/ChromecastTech.js:199]: No, eslint configs wants it with spaces.

File: src/js/tech/ChromecastTech.js (hunk 5 above)
Comment (jbreemhaar) [reply src/js/tech/ChromecastTech.js]: Yes done

File: src/js/tech/ChromecastTech.js (hunk 3 above)
Comment (jbreemhaar) [reply src/js/tech/ChromecastTech.js]: Nope, done.

File: src/js/enableChromecast.js [hunk 6]
```diff
@@ -42,8 +43,8 @@ function onChromecastRequested(player) {
 }
 
 /**
- * Adds the Chromecast button to the player's control bar, if one does not already exist,
- * then starts listening for the `chromecastRequested` event.
+ * Adds the Chromecast button to the player's control bar, if one does not already
+ * exist, then starts lister the `chromecastRequested` event.
```
Comment (MathijsR94) [inline src/js/enableChromecast.js]: Not sure what's going on here

**Review by MathijsR94 — APPROVED** [review]

**Review by sjors98 — APPROVED** [review]

File: src/js/enableChromecast.js (hunk 6 above)
Comment (jbreemhaar) [reply src/js/enableChromecast.js]: Me neither, magic .gif

**Pull request merged**